### PR TITLE
fix: fixing pdf parsing

### PR DESCRIPTION
**Pull request by jacopo-chevallard**
Our default pdf parser is Unstructured, for which we were using the 'fast' strategy, which fails in parsing some pdf. We instead use the 'auto' strategy which is more flexible and powerful.

Closes TICK-86

# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: backend/core/MegaParse/megaparse/config.py [hunk 1]
```diff
@@ -22,6 +22,6 @@ def from_yaml(cls, file_path: str):
 
 
 class MegaparseConfig(MegaparseBaseConfig):
-    strategy: str = "fast"
+    strategy: str = "auto"
```
Comment (AmineDiro) [inline backend/core/MegaParse/megaparse/config.py:25]: Need to check the impact of this on the worker

Comment (AmineDiro) [reply backend/core/MegaParse/megaparse/config.py:25]: @chloedia This was set for a reason in megaparse ? The OCR took too much time ?

**Review by AmineDiro — CHANGES_REQUESTED** [review]
Need to check the impact on infra and resource. Probably some tests to be added and ran on CI before merging

File: backend/core/MegaParse/megaparse/config.py (hunk 1 above)
Comment (AmineDiro) [reply backend/core/MegaParse/megaparse/config.py:25]: The heuristic I set up in previous project was to check the area of the "image" vs text in the page. 
Parsers like fitz return the the bbox of image, I guess unstructured should return that. Nonnative text in pdfs is represented as an image. a threshold of 0.8 worked but it depends heavily on the document being processed. 
Another heuristic is the existence of glyph encoding in the pdf. We should test that.

Comment (chloedia) [reply backend/core/MegaParse/megaparse/config.py:25]: Yes normally it already fallbacks to auto when fast is empty (according to unstructured) that s why I used it !

File: backend/api/quivr_api/modules/sync/utils/syncutils.py [hunk 2]
```diff
@@ -168,6 +168,8 @@ async def process_sync_file(
         ]:
             raise ValueError(f"Incompatible file extension for {downloaded_file}")
 
+        storage_path = f"{storage_path.split('.')[0]}.{storage_path.split('.')[-1]}"
```
Comment (AmineDiro) [inline backend/api/quivr_api/modules/sync/utils/syncutils.py:171]: we would stop at everything before the "." For example
"this.is.a.test.pdf" will be -> "this.pdf" ?

File: backend/core/MegaParse/megaparse/Converter.py [hunk 3]
```diff
@@ -320,22 +320,24 @@ async def convert(
...
             )
             md_cleaned = md_processor.process(gpt4o_cleaner=gpt4o_cleaner)
-            return LangChainDocument(
-                page_content=md_cleaned,
-                metadata={"filename": file_path.name, "type": "pdf"},
-            )
+            parsed_md = md_cleaned
+
+        if len(parsed_md) < 5 and self.strategy == "fast":
+            logger.debug(f"Switching to auto strategy for {file_path.name}")
```
Comment (AmineDiro) [inline backend/core/MegaParse/megaparse/Converter.py:333]: Should be info level

File: backend/core/MegaParse/megaparse/Converter.py [hunk 4]
```diff
@@ -320,22 +320,24 @@ async def convert(
...
+            self.strategy = "auto"
```
Comment (AmineDiro) [inline backend/core/MegaParse/megaparse/Converter.py:334]: This will set the strategy to "auto" for all subsequent parse if we use the same object 🤔

**Review by AmineDiro — APPROVED** [review]
Great

**Pull request merged**